src/main/index.ts

```diff
@@ -949,7 +950,8 @@ if (!RUNNING_WITH_COMMAND) {
             .on('click', () => {
               console.log('Clicked on OS notification');
               createOrShowMainWindow();
-              emitToWindow(MAIN_WINDOW!, 'switch-to-applet', appletId);
+              if (appletId) emitToWindow(MAIN_WINDOW!, 'switch-to-applet', appletId);
```

Comment:
What about re-using the `WeaveLocation` type here instead of taking group and appletId as separate arguments? I think it would be a bit cleaner. And I also noticed that it's actually redundant that we encode it to base64 anyway because we decode it again when handling the `switch-to-[...]` events in the frontend.

Reply:
We could probably also consolidate the event to `on-switch-to-weave-location` I think.

Reply:
Yep good, will do.